Fix CentOS/libsndfile typedef mismatch

> Thanks for patch. however im not able to to compile it.
> 
> Using Centos Stream 9.
> 
> i did cmake and make.
> 
> here is err:
> 
> ```
> ~/arancormonk/dsd-neo/tests/core/test_core_call_alert_history.c:34:1: error: conflicting types for ‘open_wav_file’; have ‘struct sf_private_tag *(char *, char *, size_t,  uint16_t,  uint8_t)’ {aka ‘struct sf_private_tag *(char *, char *, long unsigned int,  short unsigned int,  unsigned char)’}
>    34 | open_wav_file(char* dir, char* temp_filename, size_t temp_filename_size, uint16_t sample_rate, uint8_t ext) {
>       | ^~~~~~~~~~~~~
> In file included from ~/arancormonk/dsd-neo/tests/core/test_core_call_alert_history.c:8:
> ~/arancormonk/dsd-neo/include/dsd-neo/core/file_io.h:45:10: note: previous declaration of ‘open_wav_file’ with type ‘SNDFILE *(char *, char *, size_t,  uint16_t,  uint8_t)’ {aka ‘struct SNDFILE_tag *(char *, char *, long unsigned int,  short unsigned int,  unsigned char)’}
>    45 | SNDFILE* open_wav_file(char* dir, char* temp_filename, size_t temp_filename_size, uint16_t sample_rate, uint8_t ext);
>       |          ^~~~~~~~~~~~~
> ~/arancormonk/dsd-neo/tests/core/test_core_call_alert_history.c:44:1: error: conflicting types for ‘close_and_rename_wav_file’; have ‘struct sf_private_tag *(struct sf_private_tag *, const dsd_opts *, const char *, const char *, const Event_History_I *)’ {aka ‘struct sf_private_tag *(struct sf_private_tag *, const dsd_opts *, const char *, const char *, const struct Event_History_I *)’}
>    44 | close_and_rename_wav_file(struct sf_private_tag* wav_file, const dsd_opts* opts, const char* wav_out_filename,
>       | ^~~~~~~~~~~~~~~~~~~~~~~~~
> In file included from ~/arancormonk/dsd-neo/tests/core/test_core_call_alert_history.c:8:
> ~/arancormonk/dsd-neo/include/dsd-neo/core/file_io.h:49:10: note: previous declaration of ‘close_and_rename_wav_file’ with type ‘SNDFILE *(SNDFILE *, const dsd_opts *, const char *, const char *, const Event_History_I *)’ {aka ‘struct SNDFILE_tag *(struct SNDFILE_tag *, const dsd_opts *, const char *, const char *, const struct Event_History_I *)’}
>    49 | SNDFILE* close_and_rename_wav_file(SNDFILE* wav_file, const dsd_opts* opts, const char* wav_out_filename,
>       |          ^~~~~~~~~~~~~~~~~~~~~~~~~
> make[2]: *** [tests/CMakeFiles/dsd-neo_test_core_call_alert_history.dir/build.make:76: tests/CMakeFiles/dsd-neo_test_core_call_alert_history.dir/core/test_core_call_alert_history.c.o] Error 1
> make[1]: *** [CMakeFiles/Makefile2:6152: tests/CMakeFiles/dsd-neo_test_core_call_alert_history.dir/all] Error 2
> make[1]: *** Waiting for unfinished jobs....
> [ 73%] Building C object tests/CMakeFiles/dsd-neo_test_core_audio_input_rate.dir/core/test_core_audio_input_rate.c.o
> [ 73%] Linking CXX executable dsd-neo_test_core_audio_group_gate
> [ 73%] Linking CXX executable dsd-neo_test_core_talkgroup_policy
> [ 73%] Linking CXX executable dsd-neo_test_core_audio_input_rate
> [ 73%] Built target dsd-neo_test_core_talkgroup_policy
> [ 73%] Built target dsd-neo_test_core_audio_group_gate
> [ 73%] Built target dsd-neo_test_core_audio_input_rate
> make: *** [Makefile:146: all] Error 2
> ```
> 
> ```
> gcc -v
> Using built-in specs.
> COLLECT_GCC=gcc
> COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/11/lto-wrapper
> OFFLOAD_TARGET_NAMES=nvptx-none
> OFFLOAD_TARGET_DEFAULT=1
> Target: x86_64-redhat-linux
> Configured with: ../configure --enable-bootstrap --enable-host-pie --enable-host-bind-now --enable-languages=c,c++,fortran,lto --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-shared --enable-threads=posix --enable-checking=release --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-gcc-major-version-only --enable-plugin --enable-initfini-array --without-isl --enable-multilib --with-linker-hash-style=gnu --enable-offload-targets=nvptx-none --without-cuda-driver --enable-gnu-indirect-function --enable-cet --with-tune=generic --with-arch_64=x86-64-v2 --with-arch_32=x86-64 --build=x86_64-redhat-linux --with-build-config=bootstrap-lto --enable-link-serialization=1
> Thread model: posix
> Supported LTO compression algorithms: zlib zstd
> gcc version 11.5.0 20240719 (Red Hat 11.5.0-14) (GCC)
> ```
> 
> i did just freshed pulled latest refresh #146 branch main. i didn't try that on linux before, only on windows as its my main OS, i mean those previous versions. 

 _Originally posted by @mastodon987 in [#142](https://github.com/arancormonk/dsd-neo/issues/142#issuecomment-4644599310)_

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -76,6 +76,29 @@ set(_DSD_PUBLIC_HEADER_TEST_DIR
 )
 file(MAKE_DIRECTORY "${_DSD_PUBLIC_HEADER_TEST_DIR}")
 
+set(_DSD_SNDFILE_ALT_TAG_INCLUDE_DIR
+    "${CMAKE_CURRENT_BINARY_DIR}/sndfile_alt_tag_include"
+)
+file(MAKE_DIRECTORY "${_DSD_SNDFILE_ALT_TAG_INCLUDE_DIR}")
+file(
+    WRITE "${_DSD_SNDFILE_ALT_TAG_INCLUDE_DIR}/sndfile.h"
+    "#ifndef SNDFILE_H\n"
+    "#define SNDFILE_H\n"
+    "\n"
+    "typedef long long sf_count_t;\n"
+    "typedef struct SNDFILE_tag SNDFILE;\n"
+    "typedef struct SF_INFO {\n"
+    "    sf_count_t frames;\n"
+    "    int samplerate;\n"
+    "    int channels;\n"
+    "    int format;\n"
+    "    int sections;\n"
+    "    int seekable;\n"
+    "} SF_INFO;\n"
+    "\n"
+    "#endif /* SNDFILE_H */\n"
+)
+
 function(
     dsd_neo_add_public_header_smoke_test
     target_name
@@ -1393,6 +1416,34 @@ if(NOT APPLE AND NOT WIN32 AND (CMAKE_C_COMPILER_ID MATCHES "GNU|Clang"))
     add_test(
         NAME CORE_CALL_ALERT_HISTORY
         COMMAND dsd-neo_test_core_call_alert_history
+    )
+
+    add_executable(
+        dsd-neo_test_core_call_alert_history_sndfile_alt_tag
+        core/test_core_call_alert_history.c
+    )
+    target_include_directories(
+        dsd-neo_test_core_call_alert_history_sndfile_alt_tag
+        BEFORE
+        PRIVATE
+            ${_DSD_SNDFILE_ALT_TAG_INCLUDE_DIR}
+            ${PROJECT_SOURCE_DIR}/include
+    )
+    target_link_libraries(
+        dsd-neo_test_core_call_alert_history_sndfile_alt_tag
+        PRIVATE dsd-neo_core
+    )
+    target_link_options(
+        dsd-neo_test_core_call_alert_history_sndfile_alt_tag
+        PRIVATE "LINKER:--wrap=beeper"
+    )
+    set_property(
+        TARGET dsd-neo_test_core_call_alert_history_sndfile_alt_tag
+        PROPERTY EXPORT_COMPILE_COMMANDS OFF
+    )
+    add_test(
+        NAME CORE_CALL_ALERT_HISTORY_SNDFILE_ALT_TAG
+        COMMAND dsd-neo_test_core_call_alert_history_sndfile_alt_tag
     )
 endif()
 

--- a/tests/core/test_core_call_alert_history.c
+++ b/tests/core/test_core_call_alert_history.c
@@ -10,6 +10,7 @@
 #include <dsd-neo/core/state.h>
 #include <dsd-neo/core/synctype_ids.h>
 #include <dsd-neo/core/time_format.h>
+#include <dsd-neo/platform/sndfile_fwd.h>
 #include <dsd-neo/protocol/edacs/edacs_afs.h>
 #include <dsd-neo/runtime/call_alert.h>
 #include <stdint.h>
@@ -25,12 +26,10 @@
 #pragma GCC diagnostic ignored "-Wmissing-prototypes"
 #endif
 
-struct sf_private_tag;
-
 static int g_beeper_count;
 static int g_last_beeper_id;
 
-struct sf_private_tag*
+SNDFILE*
 open_wav_file(char* dir, char* temp_filename, size_t temp_filename_size, uint16_t sample_rate, uint8_t ext) {
     UNUSED(dir);
     UNUSED(temp_filename);
@@ -40,9 +39,9 @@ open_wav_file(char* dir, char* temp_filename, size_t temp_filename_size, uint16_
     return NULL;
 }
 
-struct sf_private_tag*
-close_and_rename_wav_file(struct sf_private_tag* wav_file, const dsd_opts* opts, const char* wav_out_filename,
-                          const char* dir, const Event_History_I* event_struct) {
+SNDFILE*
+close_and_rename_wav_file(SNDFILE* wav_file, const dsd_opts* opts, const char* wav_out_filename, const char* dir,
+                          const Event_History_I* event_struct) {
     UNUSED(wav_file);
     UNUSED(opts);
     UNUSED(wav_out_filename);

